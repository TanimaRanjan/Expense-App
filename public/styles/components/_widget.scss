.widget {
    background: $light-blue;
    margin-bottom: $xl-size;
    // border: 2px solid red;
    // width: 100%;
    // padding: 0;

}

// Widget header 
.widget-header{
    background: $blue;
    color:$off-white;
    padding:$m-size;
    display: flex;
    justify-content: space-between;
}

.widget-header__title {
    margin: 0;
}

.widget__message {
    border-bottom:1px solid lighten($light-blue, 10%);
    margin: 0;
    padding: $l-size;
    // width: 100%;
    text-align: center;
    color:$off-white;
}