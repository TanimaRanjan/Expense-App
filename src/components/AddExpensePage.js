import React from 'react'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'

const AddExpensePage = () => (
    <div>Add Expense Page</div>
)

export default AddExpensePage;