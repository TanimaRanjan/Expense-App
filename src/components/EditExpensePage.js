import React from 'react'
import { connect} from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense , startRemoveExpense} from '../actions/expenses';

// refactor editexpese page to be a class based componenet
// setup match 

export class EditExpensePage extends React.Component {
    onSubmit = ( expense ) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/dashboard')
    }

    onRemove =  (  ) => {
        this.props.startRemoveExpense({id: this.props.expense.id})
        this.props.history.push('/dashboard')
    }
    render() {
        return (
            <div>
                <div className='page-header'>
                    <div className='content-container'>
                        <h1 className='page-header__title'>Edit Expense</h1>
                    </div>
                </div>
                <div className='content-container'>
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className='button-layout__remove' onClick={this.onRemove}>Remove Expense</button>
                </div>
        </div>
        )
    }
}


const mapDistpatchToProps = (dispatch, props)  => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
})

const mapStateToProp = (state, props) => ({
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
})

export default connect(mapStateToProp, mapDistpatchToProps)(EditExpensePage);