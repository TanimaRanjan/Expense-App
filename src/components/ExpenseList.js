import React from 'react'
import {connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expense'
// Stateless Function 
const ExpenseList = (props) => (

        <div>
            <h2>Expense List</h2>
            {props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense} /> 
                
            })}
            
        </div>
)

const mapStateToProps = (state) => {
    return {
        expenses : selectExpenses(state.expenses, state.filters)
    }
}

const ConnectExpenseList = connect(mapStateToProps)(ExpenseList);

export default ConnectExpenseList;