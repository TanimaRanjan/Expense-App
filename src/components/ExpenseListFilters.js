
import 'react-dates/initialize';
import React from 'react'
import { connect} from 'react-redux'
import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';


export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused : null
    }
    onDatesChange = ({ startDate, endDate}) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused }))
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) => {
        if(e.target.value === 'date') {
            this.props.sortByDate()
        } else {
            this.props.sortByAmount()
        }
    }
    render() {
        return (
            <div>
            <input type="text" 
                value={this.props.filters.text}  
                onChange={this.onTextChange}
            />
            <select 
                value={this.props.filters.sortBy}
                onChange={this.onSortChange}
            >
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
            <DateRangePicker 
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false}
                startDateId={'1'}
                endDateId={'1'}
                />
        </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
        filters : state.filters
    }
}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    setEndDate:  (endDate) => dispatch(setEndDate(endDate)),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)