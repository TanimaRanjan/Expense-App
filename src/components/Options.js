import React from 'react'
import uuid  from 'uuid/v4'
import Option from './Option'

const Options = (props) => {
    // constructor(props) {
    //     super(props)
    //     this.handleRemoveAll = this.handleRemoveAll.bind(this)
    // }
    // handleRemoveAll() {
    //     console.log(this.props.options)

    // }
    
        return (
            <div>
                <button onClick={props.handleDeleteOptions}>Remove All</button>
                {props.options.length === 0 && <p>Please add an option to start</p>}
                {
                    props.options.map((option) => 
                    <Option 
                        key={uuid()} optionText={option}
                        handleDeleteOption={props.handleDeleteOption}
                        />
                    
                    )
                }
            </div>
        )
    }

    export default Options

