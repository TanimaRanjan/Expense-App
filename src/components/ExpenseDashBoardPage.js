import React from 'react'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashBoardPage = () => (
    <div>This is from my Dashboard Components
        <ExpenseListFilters></ExpenseListFilters>
        <ExpenseList></ExpenseList>
    </div>
)

export default ExpenseDashBoardPage;