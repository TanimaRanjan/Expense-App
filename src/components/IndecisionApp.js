import React from 'react'
import ReactDOM from 'react-dom'

import AddOption from './AddOption'
// import Option from './components/Option'
import Options from './Options'
import Header from './Header'
import Action from './Action'

class IndecisionApp extends React.Component {
    constructor(props) {
        super(props)
        this.handleDeleteOptions = this.handleDeleteOptions.bind(this)
        this.handleDeleteOption = this.handleDeleteOption.bind(this)
        this.handlePick = this.handlePick.bind(this)
        this.handleAddOption = this.handleAddOption.bind(this)
        this.state = {
             options : props.options
        }
    }

    componentDidMount() {
        try {
            console.log('fetching date')
            const optionsJSON = localStorage.getItem('options')
            const options = JSON.parse(optionsJSON)
            if(options) {
                this.setState(() => ({ options}))
            }
        } catch (e) {
            // Do nothing
        }
        
    }

    componentDidUpdate(preProps, prevState) {
        if(prevState.options.length !== this.state.options.length) {
            const optionJSON = JSON.stringify(this.state.options)
            localStorage.setItem('options', optionJSON)
            console.log('saving date')
        }
    }
    
    componentWillUnmount() {
        console.log('upnmount ')
    }

    handleDeleteOptions() {
        this.setState (() => ({ options: []  }))
    }
    handlePick() {
        alert(this.state.options[Math.floor(Math.random() * this.state.options.length)])

    }
    handleAddOption(option) {
        if(!option) {
            return 'Enter Valid value to add'
        } else if(this.state.options.indexOf(option) < -1)  {
            return 'This option already exists '
        } 
        this.setState((prevState) => (
            {options: prevState.options.concat(option)}) 
        )
    }
    handleDeleteOption(optionToRemove){
        console.log(optionToRemove)
     this.setState((prevState) => ({
        options:prevState.options.filter((option) =>  optionToRemove !== option
        )}))
    }

    render() {
        const title = 'Indecision App'
        const subTitle = 'Put you life in hands of computer'
        
        return (
            <div> 
                <Header title={title} subTitle={subTitle}/>
                <Action hasOptions={this.state.options.length > 0}
                handlePick={this.handlePick}
                    />
                <Options 
                    options={this.state.options} 
                    handleDeleteOptions={this.handleDeleteOptions}
                    handleDeleteOption={this.handleDeleteOption}
                />
                <AddOption 
                handleAddOption={this.handleAddOption}/>
                
            </div>
        )
    }
}

IndecisionApp.defaultProps = {
    options:[]
}




export default IndecisionApp