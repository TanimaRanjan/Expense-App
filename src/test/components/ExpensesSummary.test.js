import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import numeral from 'numeral'

test('Should correct render ExpensesSummary with 1 expense ', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should correct render ExpensesSummary with multiple expenses ', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={2353435}/>)
    expect(wrapper).toMatchSnapshot()
})