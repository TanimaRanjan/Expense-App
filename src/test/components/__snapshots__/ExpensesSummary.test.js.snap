// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correct render ExpensesSummary with 1 expense  1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h2>
</div>
`;

exports[`Should correct render ExpensesSummary with multiple expenses  1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     totalling 
    $23,534.35
  </h2>
</div>
`;
